canvas_modules/common-canvas/__tests__/context-menu/common-context-menu-test.js

```diff
@@ -36,39 +37,57 @@ import manyNodesCommentsObj from "../test_resources/json/context-menu-test_manyN
+	const ContextMenuComp = jest.requireActual(
+		"./../../src/context-menu/common-context-menu",
+	).default;
+	return <ContextMenuComp {...props} />;
+});
 
 describe("CommonContextMenu renders correctly", () => {
 
 	it("all required props should have been defined", () => {
+
 		const _contextHandler = sinon.spy();
 		const _menuDefinition = getMenuDefinition();
 		const _canvasRect = { width: 1000, height: 800, top: 0, bottom: 800, left: 0, right: 1000 };
 		const _mousePos = { x: 20, y: 20 };
-		const wrapper = mount(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
-		expect(wrapper.prop("contextHandler")).to.equal(_contextHandler);
-		expect(wrapper.prop("menuDefinition")).to.equal(_menuDefinition);
-		expect(wrapper.prop("canvasRect")).to.equal(_canvasRect);
-		expect(wrapper.prop("mousePos")).to.equal(_mousePos);
+		renderWithIntl(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
+		expectJest(mockContextMenu).toHaveBeenCalledWith({
+			"contextHandler": _contextHandler,
+			"menuDefinition": _menuDefinition,
+			"canvasRect": _canvasRect,
+			"mousePos": _mousePos
+		});
 	});
 
 	it("should render two context menu items and one divider components", () => {
 		const _contextHandler = sinon.spy();
 		const _menuDefinition = getMenuDefinition();
 		const _canvasRect = { width: 1000, height: 800, top: 0, bottom: 800, left: 0, right: 1000 };
 		const _mousePos = { x: 20, y: 20 };
-		const wrapper = mount(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
-		expect(wrapper.find(".context-menu-item")).to.have.length(2);
-		expect(wrapper.find(".context-menu-divider")).to.have.length(1);
+		const { container } = renderWithIntl(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
+
+		expect(container.getElementsByClassName("context-menu-item")).to.have.length(2);
+		expect(container.getElementsByClassName("context-menu-divider")).to.have.length(1);
 	});
 
 	it("simulates click events", () => {
 		const _contextHandler = sinon.spy();
 		const _menuDefinition = getMenuDefinition();
 		const _canvasRect = { width: 1000, height: 800, top: 0, bottom: 800, left: 0, right: 1000 };
 		const _mousePos = { x: 20, y: 20 };
-		const wrapper = mount(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
-		wrapper.find(".context-menu-item").at(0)
-			.simulate("click");
+		const wrapper = renderWithIntl(<CommonContextMenu contextHandler={_contextHandler} menuDefinition={_menuDefinition} canvasRect={_canvasRect} mousePos={_mousePos} />);
+		// <div class="context-menu-item" role ="menuitem">Do something</div>
```

Comment:
Here instead of direct HTML content as comment could you please describe as to why getByText is used below.

Reply:
I updated the comment. 
Here's my updated comment
To simulate click events, we want to click on the menu item, and one of the menu items has the text "Do something" so we use getByText to click the on that specific menu item